gene/database/postgresql.py

```diff
@@ -0,0 +1,867 @@
+        """Retrieve non-merged record. The query is pretty different, so this method
+        is broken out for PostgreSQL.
+
+        :param concept_id: ID of concept to get
+        :param case_sensitive:
+        :return: complete record object if successful
+        """
+        query = "SELECT * FROM record_lookup_view WHERE lower(concept_id) = %s;"
+        concept_id_param = concept_id.lower()
+
+        with self.conn.cursor() as cur:
+            cur.execute(query, [concept_id_param])
+            result = cur.fetchone()
+        if not result:
+            return None
+
+        gene_record = {
+            "concept_id": result[0],
+            "symbol_status": result[1],
+            "label": result[2],
+            "strand": result[3],
+            "location_annotations": result[4],
+            "locations": result[5],
+            "gene_type": result[6],
+            "aliases": result[7],
+            "associated_with": result[8],
+            "previous_symbols": result[9],
+            "symbol": result[10],
+            "xrefs": result[11],
+            "src_name": result[12],
+            "merge_ref": result[13],
+            "item_type": "identity",
+        }
+        return {k: v for k, v in gene_record.items() if v}
+
+    def _get_merged_record(
+        self, concept_id: str, case_sensitive: bool
+    ) -> Optional[Dict]:
+        """Retrieve normalized record from DB.
+
+        :param concept_id: normalized ID for the merged record
+        :param case_sensitive:
+        :return: normalized record if successful
+        """
+        concept_id = concept_id.lower()
+        query = "SELECT * FROM gene_merged WHERE lower(concept_id) = %s;"
+        with self.conn.cursor() as cur:
+            cur.execute(query, [concept_id])
+            result = cur.fetchone()
+        if not result:
+            return None
+
+        merged_record = {
+            "concept_id": result[0],
+            "symbol": result[1],
+            "symbol_status": result[2],
+            "previous_symbols": result[3],
+            "label": result[4],
+            "strand": result[5],
+            "ensembl_locations": result[6],
+            "hgnc_locations": result[7],
+            "ncbi_locations": result[8],
+            "location_annotations": result[9],
+            "ensembl_biotype": result[10],
+            "hgnc_locus_type": result[11],
+            "ncbi_gene_type": result[12],
+            "aliases": result[13],
+            "associated_with": result[14],
+            "xrefs": result[15],
+            "item_type": "merger",
+        }
+        return {k: v for k, v in merged_record.items() if v}
+
+    def get_record_by_id(self, concept_id: str, case_sensitive: bool = True,
+                         merge: bool = False) -> Optional[Dict]:
+        """Fetch record corresponding to provided concept ID
+        :param str concept_id: concept ID for gene record
+        :param bool case_sensitive:
```

Comment:
```suggestion
        :param bool case_sensitive: if true, performs exact lookup, which is
            more efficient. Otherwise, performs filter operation, which
            doesn't require correct casing.
```

Reply:
Originally I wrote separate search indexes for case-sensitive and case-insensitive searches, but then I realized that they're equally as fast, so for the sake of simplicity I got rid of the case-sensitive one and routed all searches through the case-insensitive index. That means that this param doesn't actually get used by postgres at all, but I figured it might be better to make it an explicit part of the signature since it definitely does matter for dynamo (i.e. instead of making it a kwarg)

Reply:
Ahh okay, it was just missing the description so I copied/pasted